File: MerchanCountriesAngular/MerchanCountriesAngular/ClientApp/src/app/models/country.ts
import { Name } from "./name";
import { Currencies } from "./currencies";
import { Languages } from "./languages";
import { Flags } from "./flags";
import { Entity } from "./entity";

export interface Country extends Entity {
    name: Name;
    cca2: string;
    currencies: Currencies;
    capital: string[];
    languages: Languages;
    borders: string[];
    population: number;
    timezones: string[];
    flags: Flags;
}

File: MerchanCountriesApi/MerchanCountriesApi.Domain/Country.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text.Json.Serialization;
using System.Threading.Tasks;
using MerchanCountriesApi.Core.Models;
using Newtonsoft.Json;

namespace MerchanCountriesApi.Domain
{
    public class Country : Entity
    {
        public Country(Guid id) : base(id)
        {
        }

        [JsonPropertyName("name")]
        public Name Name { get; set; }

        [JsonPropertyName("cca2")]
        public string Cca2 { get; set; }

        [JsonPropertyName("currencies")]
        public Currencies Currencies { get; set; }

        [JsonPropertyName("capital")]
        public List<string> Capital { get; set; }

        [JsonPropertyName("languages")]
        public Languages Languages { get; set; }

        [JsonPropertyName("borders")]
        public List<string> Borders { get; set; }

        [JsonPropertyName("population")]
        public int Population { get; set; }

        [JsonPropertyName("timezones")]
        public List<string> Timezones { get; set; }

        [JsonPropertyName("flags")]
        public Flags Flags { get; set; }
    }
}